docs/configuration-format.md

```diff
@@ -0,0 +1,157 @@
+# OpenTelemetry Collector Configuration Format
+
+The effective configuration that defines runtime behavior of the collector is calculated from 2 sources: the Default configuration which is hard-coded and the User-defined configuration which supplied by the user via various configuration sources. Default configuration defines reasonable defaults that most users will likely want to use. User-defined configuration can override the defaults and specify additional configuration. The effective configuration is calculated by merging Default and User-defined configurations (see merging rules in a separate section below).
+
+The top-level entities in the configuration file are receivers, processors, exporters, pipeline (more about each below in separate sections). Here is the top-level structure of the configuration:
+
+```yaml
+receivers:
+  # map of receivers
+processors:
+  # map of processors
+exporters:
+  # map of exporters
+extensions:
+  # map of extensions
+pipelines:
+  # map of pipelines
+```
+
+## Receivers
+
+Each receiver has a name which is defined by the key of the mapping under receivers top level-key. The name should be in the form `type[/name]`, where type is the type of the receiver (e.g. `otlp`), and `name` suffix is optionally appended after a forward slash to ensure uniqueness of the full name of the receiver. It is not allowed to define more than one receiver with the same name.
+
+If more than one receiver is defined that listens on the same port it is treated as a fatal configuration error (note: it is valid for 2 receivers to listen on the same port number if they bind to different network interfaces).
+
+More than one pipeline can be associated with the same receiver definition. In that case the data from that receiver is copied (fanned out) to all associated pipelines. 
+The definition of receiver is specific for each type however at the minimum it will include a port number and the network interface address to bind to (defaults to `127.0.0.1`).
+
+There are 2 possible configuration structure for a receiver. First structure is for receivers that support only one protocol:
+
+```yaml
+receivers:
+  <receiver name>:
+    endpoint: <network interface and port to bind to, address:port>
+    enabled: <boolean, defaults to true>
```

Comment:
this is no longer a standard to have "enabled". Changed this between design and implementation.

Reply:
Removed